mitodl/open-discussions
# Limit max depth of comments

#### What are the relevant tickets?
Fixes #238 
Fixes #269 

#### What's this PR do?
 - Limits the comment depth in the UI to 6: this prevents comments deeper than this from being rendered and also removes the 'Reply' button. The user can still add more comments via REST API, they won't be able to see them though.
 - Deletes the GET method from the comment detail view since it's not used anywhere and leaving it would complicate the work of limiting comment depth.
 - Adds code to prevent any `MoreComments` from becoming part of the tree. These extra comments will just be omitted. We should not be dealing with this anyway except in the case of something unexpected on the backend

#### How should this be manually tested?
Make a whole bunch of comments and verify that you can't reply to them if depth is six. You can tweak this value via environment variable.
